pkg/broker/ingress/handler.go

```diff
@@ -134,6 +140,16 @@ func (h *handler) ServeHTTP(response nethttp.ResponseWriter, request *nethttp.Re
 		return
 	}
 
+	h.eventHops.UpdateRemainingHops(event, defaultEventHopsLimit)
```

Comment:
Looks like the hops is decremented twice: once in the deliver  processor and once here. 

I think we only need to `GetRemainingHops` in ingress. Or what if we let the delivery processor to drop events that exceeds the hop limit? That way ingress doesn't need to do anything.

Reply:
No, in the deliver processor it sets the original hops without decrementing. The `UpdateRemainingHops` function only decrements hops when the event already has a Hops value. But the event reply won't have that (we also attempt to remove it no matter if it presents).

`UpdateRemainingHops` in ingress is needed (with the current approach) in order to set the initial hops. Doing it only in delivery processor is an interesting idea. I actually didn't see a problem but maybe there are some caveats. @Harwayne thoughts?

Reply:
I had two reasons for putting it in the ingress in the initial Broker implementation:
1. The event is dropped before being duplicated N times (for N Triggers).
1. It was a quick way to help prevent people from accidentally sending events directly to the internal Channel.

1 is not an issue for this Broker design. 2 could be, depending on how you want to think about it. There is no Channel object, but people could still send events directly to the PubSub Topic. By mandating that this extension is present, it makes it harder to do so by mistake (although a malicious actor would have no problem). 

I also liked the fact that all the 'drop the event' logic was in a single place. As it turned out, we never did any other form of filtering, so this was a moot point.

If nothing else, by doing it in the Ingress, we prevent another hop through PubSub, right? Whether that is worth the extra complexity of touching the attribute in both components is up to you.

Reply:
I'm thinking we would drop the replied event right before forwarding to the ingress. It's a step before the ingress and shouldn't incur additional pubsub hop. If you don't see any other problems, I think I will proceed with having the logic in a single place.